src/AerospikeCache.php

```diff
@@ -0,0 +1,133 @@
+<?php
+declare(strict_types=1);
+
+namespace Lmc\AerospikeCache;
+
+use Aerospike;
+use Symfony\Component\Cache\Adapter\AbstractAdapter;
+use Symfony\Component\Cache\Exception\CacheException;
+
+class AerospikeCache extends AbstractAdapter
+{
+    private const WRAPPER_NAME = 'data';
+
+    /** @var Aerospike */
+    private $aerospike;
+
+    /** @var string $aerospikeNamespace */
+    private $namespace;
+
+    /** @var string $set */
+    private $set;
+
+    public function __construct(
+        \Aerospike $aerospike,
+        string $namespace,
+        string $set = 'cache',
+        string $cacheNamespace = '',
+        int $defaultLifetime = 0
+    ) {
+        $this->aerospike = $aerospike;
+        $this->namespace = $namespace;
+        $this->set = $set;
+        parent::__construct($cacheNamespace, $defaultLifetime);
+    }
+
+    protected function doFetch(array $ids): array
+    {
+        $result = [];
+
+        $keys = $this->initializeKeysForAerospike($ids);
+
+        $this->aerospike->getMany($keys, $records);
+
+        foreach ($records as $record) {
+            if ($record['metadata'] !== null) {
+                $result[$record['key']['key']] = isset($record['bins'][self::WRAPPER_NAME]) ?
+                    $record['bins'][self::WRAPPER_NAME] : null;
+            }
+        }
+
+        return $result;
+    }
+
+    private function initializeKeysForAerospike(array $ids): array
+    {
+        return array_reduce(
+            $ids,
+            function (array $values, $id) {
+                $values[] = $this->createKey($id);
+
+                return $values;
+            },
+            []
+        );
+    }
+
+    protected function doHave($id)
+    {
+        $statusCode = $this->aerospike->get($this->createKey($id), $read);
+
+        return $statusCode === \Aerospike::OK;
+    }
+
+    protected function doClear($namespace = ''): bool
+    {
+        if ($namespace === '') {
+            $statusCode = $this->aerospike->truncate($this->namespace, $this->set, 0);
+        } else {
+            $clearNamespace = function ($record) use ($namespace): void {
+                if ($namespace === mb_substr($record['key']['key'], 0, mb_strlen($namespace))) {
+                    $status = $this->aerospike->remove($record['key']);
+                    if (!$this->isStatusOkOrNotFound($status)) {
+                        throw new CacheException($this->aerospike->error());
```

Comment:
Why `doClear` throws an Exception on failure but `doDelete` just returns false - whats the difference?

Reply:
The doClear throws exception because it is in a callback. The returned value wont get throuth the scan method. 
I added a try catch block that will intercept the exception and will return a false in the catch statement.